File: src/main/java/com/FRS/main/entities/Devis.java
package com.FRS.main.entities;

import java.io.Serializable;
import java.sql.Date;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.PrimaryKeyJoinColumn;

@Entity
@DiscriminatorValue( value="devis" )
@PrimaryKeyJoinColumn(referencedColumnName="Rid")
public class Devis extends Reparation {
	
private Long DEid;
private String Commentaire;

public Devis() {
	super();
	
}


public Devis( Client client, Boutique boutique, Date date_heure, int id_Créateur, double tarif, int dEid,
		String commentaire) {
	super(client, boutique, date_heure, id_Créateur, tarif);

	Commentaire = commentaire;
}


public String getCommentaire() {
	return Commentaire;
}

public void setCommentaire(String commentaire) {
	Commentaire = commentaire;
}


public Long getDEid() {
	return DEid;
}


public void setDEid(Long dEid) {
	DEid = dEid;
}


}

File: src/main/resources/static/gesrep1/src/app/enteties/Devis.ts
import { Personnel } from "./Personnel";
import { Boutique } from "./Boutique";
import { Client } from "./Client";
import { Type_Panne } from "./Type_Panne";

export class Devis {
    DEid:number;
    commentaire:String;
    type_panne:Type_Panne;
    client:Client; 
    boutique: Boutique;
     date_heure:Date;
    createur:Personnel; 
    tarif:number;
}